12-factor.md

```diff
@@ -0,0 +1,80 @@
+# 12 Factor App : Hackney Development Standards
+
+The Twelve Factor App Methodology is suggested by developers for smoothly working and delivering Software as a Service (SaaS) Applications or Web Apps with a focus on Microservices. 
+
+## 1. One codebase tracked in revision control, many deploys
+
+- *As a developer I should put my application code in GitHub so that I can showcase my work, get review and feedback from a wider group than my colleagues.*
+
+For example :https://github.com/LBHackney-IT/LBHTenancyAPI
+
+You should always have one repository for an individual application to ease CI/CD pipelines. Also you can have multiple repositories for a microservice.
+The codebase is the same across all deployments, although different versions may be active in each deployment. For example, a developer has some commits not yet deployed to staging; staging will have some commits which are not yet deployed to production environment. But they all share the same codebase, thus making them identifiable as different deployments of the same application. Working in the open allows us to have feedback from a wider audience thus giving an opportunity to improve the code quality.
```

Comment:
```suggestion
The codebase is the same across all deployments, although different versions may be active in each deployment. For example, a developer has some commits not yet deployed to staging; staging will have some commits which are not yet deployed to the production environment. But they all share the same codebase, thus making them identifiable as different deployments of the same application. Deployments should only be normally made from the CI, not from developers machine.  Working in the open allows us to have feedback from a wider audience thus giving an opportunity to improve the code quality.
```

Its worth metioing that deploys should not be done localy hence i have added the suggestion `Deployments should only be made from the CI, not from developers machine.`